.github/workflows/docker_snapshot.yml

```diff
@@ -0,0 +1,81 @@
+
+on:
+  push:
+    branches:
+      - 'master'
+      - '[0-9]+.[0-9]+.z'
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'hazelcast'
+    steps:
+      - name: Checkout hazelcast-snapshot repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast-dockerfiles/hazelcast-snapshot
+          ref: master
+          path: hazelcast-snapshot
+      - name: Checkout hazelcast-enterprise-snapshot repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast-dockerfiles/hazelcast-enterprise-snapshot
+          ref: master
+          path: hazelcast-enterprise-snapshot
+      - name: Checkout hazelcast repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast/hazelcast
+          path: hazelcast
+      - name: Checkout hazelcast-enterprise repo
+        uses: actions/checkout@v2
+        with:
+          repository: hazelcast/hazelcast-enterprise
+          ref: ${{ github.ref }}
+          path: hazelcast-enterprise
+          token: ${{ secrets.GH_PAT }}
+
+      - name: Set up Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker Hub
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+
+      - name: Build hazelcast with Maven
+        run: |
+          mvn -f hazelcast/pom.xml -B clean install -DskipTests
+          rm hazelcast/hazelcast-all/target/*-sources.jar
+          cp hazelcast/hazelcast-all/target/hazelcast-all-*.jar hazelcast-all.jar
+
+      - name: Docker Build and Push hazelcast-snapshot image
+        run: |
+          BRANCH="${GITHUB_REF#refs/heads/}"
+          TAGS="--tag devopshazelcast/hazelcast-snapshot:${BRANCH}"
+          if [[ "${BRANCH}" == "master" ]]; then
+            TAGS="${TAGS} --tag devopshazelcast/hazelcast-snapshot:latest"
+          fi
+          docker buildx build --push -f hazelcast-snapshot/Dockerfile $TAGS .
+
+      - name: Build hazelcast-enterprise with Maven
+        run: |
+          mvn -f hazelcast-enterprise/pom.xml -B clean install -DskipTests
+          rm hazelcast-enterprise/hazelcast-enterprise-all/target/*-tests.jar
+          cp hazelcast-enterprise/hazelcast-enterprise-all/target/hazelcast-enterprise-all-*.jar hazelcast-enterprise-all.jar
+
+      - name: Docker Build and Push hazelcast-enterprise-snapshot image
+        run: |
+          BRANCH="${GITHUB_REF#refs/heads/}"
+          TAGS="--tag devopshazelcast/hazelcast-enterprise-snapshot:${BRANCH}"
+          if [[ "${BRANCH}" == "master" ]]; then
+            TAGS="${TAGS} --tag devopshazelcast/hazelcast-enterprise-snapshot:latest"
+          fi
+          docker buildx build --push -f hazelcast-enterprise-snapshot/Dockerfile $TAGS .
```

Comment:
I don't quite understand why we update hazelcast-enterprise image when hazelcast repo gets new commits. There is already an another action for hazelcast-enterprise repo (at least there is a PR).

Reply:
Hazelcast Enterprise embeds code from this repo. So if we change something in this one we should rebuild both.